admin_manual/installation/source_installation.rst

```diff
@@ -38,86 +38,118 @@ Apache and MariaDB, using `the ownCloud .tar archive
 
-* php5 (>= 5.6)
-* PHP module ctype
-* PHP module dom
-* PHP module GD
-* PHP module iconv
-* PHP module JSON
-* PHP module libxml (Linux package libxml2 must be >=2.7.0)
-* PHP module mb multibyte
-* PHP module posix
-* PHP module SimpleXML
-* PHP module XMLWriter
-* PHP module zip
-* PHP module zlib
+If you are using a Linux distribution it should have packages for all the
+required extensions. You can check the presence of a module by typing ``php -m
+| grep -i <module_name>``. If you get a result, the module is present.
 
-Database connectors (pick the one for your database:)
+Required
+^^^^^^^^
 
-* PHP module sqlite (>= 3, usually not recommended for performance reasons)
-* PHP module pdo_mysql (MySQL/MariaDB)
-* PHP module pgsql (requires PostgreSQL >= 9.0)
+* php5 (>= 5.6)
+* ctype
+* dom
+* GD
+* iconv
+* JSON
+* libxml (Linux package libxml2 must be >=2.7.0)
+* mb multibyte
+* posix
+* SimpleXML
+* XMLWriter
+* zip
+* zlib
 
-*Recommended* packages:
+Database connectors (pick the one for your database:)
 
-* PHP module curl (highly recommended, some functionality, e.g. HTTP user
-  authentication, depends on this)
-* PHP module fileinfo (highly recommended, enhances file analysis performance)
-* PHP module bz2 (recommended, required for extraction of apps)
-* PHP module intl (increases language translation performance and fixes sorting 
-  of non-ASCII characters)
-* PHP module mcrypt (increases file encryption performance)
-* PHP module openssl (required for accessing HTTPS resources)
+* pdo_mysql (MySQL/MariaDB)
+* pgsql (requires PostgreSQL >= 9.0)
+* sqlite (>= 3, usually not recommended for performance reasons)
 
-Required for specific apps:
+Required For Specific Apps
+^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-* PHP module ldap (for LDAP integration)
-* PHP module ftp (for FTP storage / external user authentication)
-* PHP module imap (for external user authentication)
-* PHP module smbclient  (SMB/CIFS integration)
+* ftp (for FTP storage / external user authentication)
+* imap (for external user authentication)
+* ldap (for LDAP integration)
+* smbclient  (SMB/CIFS integration)
   
 .. note:: SMB/Windows Network Drive mounts require the PHP module smbclient version 0.8.0+; see
   :doc:`../configuration_files/external_storage/smb`.
 
-Recommended for specific apps (*optional*):
-
-* PHP module exif (for image rotation in pictures app)
-* PHP module gmp (for SFTP storage)
-
-For enhanced server performance (*optional*) select one of the following 
-memcaches:
-
-* PHP module apc
-* PHP module apcu
-* PHP module memcached
-* PHP module redis (>= 2.2.6+, required for Transactional File Locking)
+Optional
+^^^^^^^^
+
+========= =====================================================================
+Extension Reason
+========= =====================================================================
+bz2       Required for extraction of apps
+curl      Highly recommended, as some functionality, such as HTTP user 
```

Comment:
Leave it with me. I'm hoping to be able to do this increasingly efficiently the more I come to know the respective codebases.